sdk/textanalytics/ai-text-analytics/src/transforms.ts

```diff
@@ -0,0 +1,346 @@
+ * is user-friendly.
+ *
+ * @param sentence - The sentence sentiment object to be converted.
+ * @param response - The entire response returned by the service.
+ * @returns The user-friendly sentence sentiment object.
+ * @internal
+ */
+function convertGeneratedSentenceSentiment(
+  { targets, assessments: _, ...rest }: GeneratedSentenceSentiment,
+  sentences: GeneratedSentenceSentiment[]
+): SentenceSentiment {
+  return {
+    ...rest,
+    opinions:
+      targets?.map(
+        // eslint-disable-next-line @typescript-eslint/no-shadow
+        ({ relations, ...rest }: SentenceTarget): Opinion => ({
+          target: rest,
+          assessments: relations
+            .filter((relation) => relation.relationType === "assessment")
+            .map((relation) => convertTargetRelationToAssessmentSentiment(relation, sentences)),
+        })
+      ) ?? [],
+  };
+}
+
+/**
+ * Converts a target relation object returned by the service to an assessment
+ * sentiment object where JSON pointers in the former are realized in the
+ * latter.
+ *
+ * @param targetRelation - The target relation object to be converted.
+ * @param response - The entire response returned by the service.
+ * @returns The user-friendly assessment sentiment object.
+ * @internal
+ */
+function convertTargetRelationToAssessmentSentiment(
+  targetRelation: TargetRelation,
+  sentences: GeneratedSentenceSentiment[]
+): AssessmentSentiment {
+  const assessmentPtr = targetRelation.ref;
+  const assessmentIndex: AssessmentIndex = parseAssessmentIndex(assessmentPtr);
+  const assessment: SentenceAssessment | undefined =
+    sentences?.[assessmentIndex.sentence].assessments?.[assessmentIndex.assessment];
+  if (assessment !== undefined) {
+    return assessment;
+  } else {
+    throw new Error(`Pointer "${assessmentPtr}" is not a valid Assessment pointer`);
+  }
+}
+
+/**
+ * @internal
+ */
+export function toEntityLinkingResult(
+  documents: TextDocumentInput[],
+  results: GeneratedEntityLinkingResult
+): EntityLinkingResult[] {
+  return transformDocumentResults(documents, results);
+}
+
+/**
+ * @internal
+ */
+export function toKeyPhraseExtractionResult(
+  documents: TextDocumentInput[],
+  results: GeneratedKeyPhraseExtractionResult
+): KeyPhraseExtractionResult[] {
+  return transformDocumentResults(documents, results);
+}
+
+/**
+ * @internal
+ */
+export function toEntityRecognitionResult(
+  documents: TextDocumentInput[],
+  results: GeneratedEntityRecognitionResult
+): EntityRecognitionResult[] {
+  return transformDocumentResults(documents, results);
+}
+
+/**
+ * @internal
+ */
+export function transformActionResult(
+  actionName: AnalyzeActionName,
+  input: TextDocumentInput[] | LanguageDetectionInput[],
+  response: AnalyzeResponse
+):
+  | EntityLinkingResult[]
+  | EntityRecognitionResult[]
+  | KeyPhraseExtractionResult[]
+  | SentimentAnalysisResult[]
+  | LanguageDetectionResult[] {
```

Comment:
Youve used generic inference before to tighten these functions, could you do the same here?

Reply:
reused `AnalyzeResult` as the return type.